### PR TITLE
add saiba build to preview action

### PR DESCRIPTION
**Pull request by vabarbosa**
## Changes

this PR includes `saiba` into the preview workflow. this is a pre-requisites to getting the PR previews for `saiba` repo

https://github.com/Qiskit/saiba/pull/211 depends on this PR

related https://github.com/Qiskit/saiba/issues/195, https://github.com/Qiskit/saiba/issues/197

## Implementation details

the preview workflow is added to include the build and deployment of `saiba` PR branch

## How to read this PR

review updates to preview workflow script

### CODE REVIEW

File: .github/workflows/code-engine-preview.yml [hunk 1]
```diff
@@ -116,6 +119,17 @@ jobs:
...
+      - name: Build and push to IBM registry for saiba
+        if: github.repository == 'Qiskit/saiba'
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          tags: ${{ env.DOCKER_IMAGE_TAG }}
+          cache-from: type=registry,ref=${{ env.DOCKER_IMAGE_TAG }}
+          cache-to: type=inline
+          build-args: |
+            PUBLIC_API_BASE=inputs.public_api_base
```
Comment (Tansito) [inline .github/workflows/code-engine-preview.yml:131]: Probably we don't need this if not that taking into account the `environment` the url will be set in `nuxt.config` [configuration file](https://github.com/Qiskit/saiba/blob/main/nuxt.config.ts#L40)?

Comment (vabarbosa) [reply .github/workflows/code-engine-preview.yml:131]: the default in `nuxt.config` is `http://localhost:8055`? are you suggesting we change this default?

Comment (abdonrd) [reply .github/workflows/code-engine-preview.yml:131]: I think it should be `NUXT_PUBLIC_API_BASE`.

Comment (vabarbosa) [reply .github/workflows/code-engine-preview.yml:131]: i set it to `NUXT_PUBLIC_API_BASE` in the dockerfile: https://github.com/Qiskit/saiba/blob/9e1c8514cb56b4fd20af454ba23f0e5c52838779/Dockerfile#L22
but yeah, i should probably keep it consistent throughout

Comment (abdonrd) [reply .github/workflows/code-engine-preview.yml:131]: Ooh, got it! Your are right.
No problem from my side to keep it as it is.

**Comment by vabarbosa** [comment]
closing ... as no longer needed...

File: .github/workflows/code-engine-preview.yml (hunk 1 above)
Comment (vabarbosa) [reply .github/workflows/code-engine-preview.yml:131]: took approach suggested by @Tansito so no longer need this PR

**Pull request closed without merging**